smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddHttpApiKeyAuthPlugin.java

```diff
@@ -123,6 +124,7 @@ public void writeAdditionalFiles(
         writerFactory.accept(
                 Paths.get(CodegenUtils.SOURCE_FOLDER, "middleware", INTEGRATION_NAME, "index.ts").toString(),
                 writer -> {
+                        writer.addDependency(TypeScriptDependency.AWS_SDK_UTIL_MIDDLEWARE);
```

Comment:
If I look at [http-api-key-auth.ts](https://github.com/awslabs/smithy-typescript/blob/main/smithy-typescript-codegen/src/main/resources/software/amazon/smithy/typescript/codegen/integration/http-api-key-auth.ts) I don't see @aws-sdk/util-middleware used. What am I missing?

Reply:
True. It is because we are working on the bearer token configuration and that needs a Promise to passed instead of a static value.

Reply:
I have pushed the changes to `http-api-key-auth.ts` file now.